File: Dddml.Wms.JavaCommon/src/org/dddml/wms/domain/InOutStateEventIdFlattenedDto.java
package org.dddml.wms.domain;

import java.util.HashMap;
import java.util.Map;

public class InOutStateEventIdFlattenedDto
{


    public static final String[] propertyNames = new String[]{
            "documentNumber",
            "version",
        };

    public static final String[] propertyTypes = new String[]{
            "String",
            "Long",
        };

    public static final Map<String, String> propertyTypeMap;

    static {
        propertyTypeMap = new HashMap<String, String>();
        for (int i = 0; i < propertyNames.length; i++ ) {
            propertyTypeMap.put(propertyNames[i], propertyTypes[i]);
        }
    }
	
    private InOutStateEventId value;

    public InOutStateEventIdFlattenedDto()
    {
        this(new InOutStateEventId());
    }

    public InOutStateEventIdFlattenedDto(InOutStateEventId value)
    {
        this.value = value;
    }

    public InOutStateEventId toInOutStateEventId()
    {
        return this.value;
    }


    public String getDocumentNumber()
    {
        return this.value.getDocumentNumber();
    }

    public void setDocumentNumber(String documentNumber)
    {
        this.value.setDocumentNumber(documentNumber);
    }

    public Long getVersion()
    {
        return this.value.getVersion();
    }

    public void setVersion(Long version)
    {
        this.value.setVersion(version);
    }


    @Override
    public boolean equals(Object obj)
    {
        if (obj == this) {
            return true;
        }
        if (obj == null || obj.getClass() != this.getClass()) {
            return false;
        }

        InOutStateEventIdFlattenedDto other = (InOutStateEventIdFlattenedDto)obj;
        return value.equals(other.value);
    }

    @Override
    public int hashCode()
    {
        return value.hashCode();
    }

}

File: Dddml.Wms.HttpServices.PhpClient/Dddml/Wms/Domain/InOutStateEventIdFlattenedDto.php
<?php

namespace Dddml\Wms\Domain;

use JMS\Serializer\Annotation\Type;
use Dddml\Serializer\Type\Long;
use Dddml\FlattenedDtoInterface;

class InOutStateEventIdFlattenedDto implements FlattenedDtoInterface
{

    const PROPERTIES = [
            'documentNumber' => 'string',
            'version' => 'Long',
        ];

    public static function getPropertyNames()
    {
        return array_keys(static::PROPERTIES);
    }

    public static function getPropertyTypes()
    {
        return array_values(static::PROPERTIES);
    }

	
    /**
     * @var InOutStateEventId
     */
    private $value;

    /**
     * @param InOutStateEventId $value
     */
    public function __construct(InOutStateEventId $value = null)
    {
        if ($value) {
            $this->value = $value;
        } else {
            $this->value = new InOutStateEventId();
        }
    }

    /**
     * @return InOutStateEventId
     */
    public function toInOutStateEventId()
    {
        return $this->value;
    }

    /**
     * @return string
     */
    public function getDocumentNumber()
    {
        return $this->value->getDocumentNumber();
    }

    /**
     * @param string $documentNumber
     */
    public function setDocumentNumber($documentNumber)
    {
        $this->value->setDocumentNumber($documentNumber);
    }

    /**
     * @return Long
     */
    public function getVersion()
    {
        return $this->value->getVersion();
    }

    /**
     * @param Long $version
     */
    public function setVersion($version)
    {
        $this->value->setVersion($version);
    }

    /**
     * @return string
     */
    public function toString()
    {
        $pValues = [
            $this->getDocumentNumber(),
            $this->getVersion(),
        ];
        return implode(',', $pValues);
    }

    /**
     * @param string $data
     *
     * @return InOutStateEventIdFlattenedDto
     */
    public function fromString($data)
    {
        $pValues = explode(',', $data);
        $this->setDocumentNumber($pValues[0]);
        $this->setVersion($pValues[1]);
        return $this;
    }

}

